src/components/Articles.jsx

```diff
@@ -2,27 +2,39 @@ import { useEffect, useState } from "react";
 import { getAllArticles } from "../api";
 import { Link } from "react-router-dom";
 import ArticleCard from "./ArticleCard";
+import { useSearchParams } from "react-router-dom";
 
 export default function Articles({ isLoading, setIsLoading }) {
   const [articles, setArticles] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const topicQuery = searchParams.get("topic");
 
   useEffect(() => {
-    getAllArticles()
+    const fetchArticles = topicQuery
+      ? getAllArticles(topicQuery)
+      : getAllArticles();
```

Comment:
fell this is unnecessary, you can invoke with topicQuery, but it can be undefined and if any conditional logic is require, do it in api.js.